Sources/FoundationExtensions/JSONDecoder+Extensions.swift

```diff
@@ -33,41 +33,63 @@ enum CodableError: Error, CustomStringConvertible {
 
 extension JSONDecoder {
 
-    /// Decodes a top-level value of the given type from the given Dictionary.
+    /// Decodes a top-level value of the given type from the given Data containing a JSON representation of `type`.
     ///
     /// - Parameters:
     ///   - type: The type of the value to decode. The default is `T.self`.
-    ///   - dictionary: The dictionary to decode from.
-    ///   - keyDecodingStrategy: The strategy to use for automatically changing the
-    ///   value of keys before decoding. The default is `useDefaultKeys`.
+    ///   - data: The data to decode from.
     ///   - dateDecodingStrategy: The strategy to use for decoding `Date` values. The default is `deferredToDate`.
     ///   - dataDecodingStrategy: The strategy to use for decoding `Data` values. The default is `deferredToData`.
     /// - Returns: A value of the requested type.
     /// - throws: `CodableError` or `DecodableError` if the data is invalid or can't be deserialized.
     func decode<T: Decodable>(
         _ type: T.Type = T.self,
-        dictionary: [String: Any],
-        keyDecodingStrategy: KeyDecodingStrategy = .useDefaultKeys,
+        jsonData: Data,
         dateDecodingStrategy: DateDecodingStrategy = .deferredToDate,
-        dataDecodingStrategy: DataDecodingStrategy = .deferredToData
+        dataDecodingStrategy: DataDecodingStrategy = .deferredToData,
+        logErrors: Bool = true
     ) throws -> T {
-        self.keyDecodingStrategy = keyDecodingStrategy
```

Comment:
This `decode` method mutates the instance it's called on, which isn't a great practice. At least the `keyDecodingStrategy` is no longer changed and we rely on a single `defaultJsonDecoder` everywhere. I should look at the `date` and `data` decoding strategies, we might be able to just set them on `defaultJsonDecoder` too.

Reply:
Done: #1432